### PR TITLE
Build and test on java 11

### PR DESCRIPTION
**Pull request by lbergelson**
### Description
This is an extension of #1269 which resolves some problems with the build in gradle 5+

### Checklist

- [ ] Code compiles correctly
- [ ] New tests covering changes and new functionality
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

**Comment by lbergelson** [comment]
@cmnbroad Could you take a look at this when you get a chance?  It has a bunch of gradle changes because the `maven` publishing plugin broke when we moved to gradle 5 and rather than figure out what was going wrong I updated it to use the `maven-publish` plugin instead which is the new recommended one.

File: build.gradle [hunk 1]
```diff
@@ -141,92 +142,79 @@ task testExternalApis(type: Test) {
...
+                url = 'http://samtools.github.io/htsjdk/'
 
                 developers {
                     developer {
-                        id 'picard'
-                        name 'Picard Team'
-                        url 'http://broadinstitute.github.io/picard'
+                        id = 'picard'
+                        name = 'Picard Team'
+                        url = 'http://broadinstitute.github.io/picard'
```
Comment (cmnbroad) [inline build.gradle]: Unrelated and tangential, but this seems like ancient history. And if you look there, you see nothing about htsjdk.

File: build.gradle [hunk 2]
```diff
@@ -141,92 +142,79 @@ task testExternalApis(type: Test) {
...
-    archives sourcesJar
+    archiveClassifier.set('sources')
 }
 
 /**
  * Sign non-snapshot releases with our secret key.  This should never need to be invoked directly.
  */
 signing {
-    required { isRelease && gradle.taskGraph.hasTask("uploadArchives") }
+    required { isRelease && gradle.taskGraph.hasTask("publishHtsjdkPublicationToMavenRepository") }
```
Comment (cmnbroad) [inline build.gradle]: Is the uppercase 'H' in "Htsjdk" significant? It seems to work ok locally (it winds up in "htsjdk"ok).

File: gradlew [hunk 3]
```diff
@@ -28,7 +28,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m"'
```
Comment (cmnbroad) [inline gradlew]: Wondering what triggered this change ?

File: build.gradle (hunk 1 above)
Comment (lbergelson) [reply build.gradle]: Yeah, I noticed that too but decided to leave it for later...  I'll open a follow up to change it.  Not sure to what though...

File: build.gradle (hunk 2 above)
Comment (lbergelson) [reply build.gradle]: I don't believe it's meaningful.  I think it's case insensitive but the examples on the gradle website use camel case so I did as well.

File: gradlew (hunk 3 above)
Comment (lbergelson) [reply gradlew]: Rebuilding the wrapper triggered it.  I think they reduced gradle's memory requirement so they added a small default one.

**Pull request merged**